lib/scripting/objects/pinball-backend/rom-fetcher.spec.ts

```diff
@@ -0,0 +1,119 @@
+
+	before(() => {
+		globalAny.fetch = fetch;
+	});
+
+	after(() => {
+		globalAny.fetch = undefined;
+	});
+
+	it('should not find a rom entry', () => {
+		return downloadGameEntry('XXX')
+			.catch((error) => {
+				expect(error.message).match(/GAME_ENTRY_NOT_FOUND_XXX/);
+			});
+	});
+
+	it('should successful download ROM', () => {
+		const nockScopeA = nock('https://api.vpdb.io')
+			.get('/v1/games/mm/roms/')
+			.reply(200, VPDB_GAME_ENTRY_JSON);
+
+		const nockScopeB= nock('https://storage.vpdb.io')
+			.get('/files/p2b9gpvd1.zip/mm_1_09c.bin')
+			.reply(200, MOCK_ROM);
+
+		return downloadGameEntry('mm_109c')
+			.then((answer: LoadedGameEntry) => {
+				expect(answer.romFile).to.deep.equal(new Uint8Array([ 77, 79, 67, 75, 82, 79, 77 ]));
+				expect(answer.wpcDbEntry.name).to.equal('WPC-95: Medieval Madness');
+				expect(nockScopeA.isDone()).to.equal(true);
+				expect(nockScopeB.isDone()).to.equal(true);
+			});
+	});
+
+	it('should handle invalid rom list, option 1', () => {
+		const nockScopeA = nock('https://api.vpdb.io')
+			.get('/v1/games/mm/roms/')
+			.reply(200, { foo: 'bar' });
+
+		return downloadGameEntry('mm_109c')
```

Comment:
this should work too:

```ts
expect(async () => await downloadGameEntry('mm_109c')).to.throw(/VPDB_INVALID_ANSWER/);
```

If not, have a look at [chai-as-promised](https://www.chaijs.com/plugins/chai-as-promised/)